x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/integrity/RepositoryVerifyIntegrityIT.java

```diff
@@ -344,16 +344,21 @@ public void testCorruption() throws IOException {
                 ? equalTo(testContext.indexNames().size())
                 : lessThan(testContext.indexNames().size())
         );
-        assertThat(anomalies, not(empty()));
+        // Missing shard generation file is automatically repaired based on the shard snapshot files.
+        // See also BlobStoreRepository#buildBlobStoreIndexShardSnapshots
+        assertThat(anomalies, corruptedFileType == RepositoryFileType.SHARD_GENERATION ? empty() : not(empty()));
```

Comment:
Alternatively, we can add a flag to make `buildBlobStoreIndexShardSnapshots` not repair missing shard gen file so that the verification can fail. I am not sure whether it is necessary since this would return warnings to users for something we already have a workaround, i.e. no action is required on the user side. In addition, the warning message about the repairing is logged in this case and I think that is all we need for this type of issue.